docs/ADDING_NEW_EXCHANGE.md

```diff
@@ -0,0 +1,461 @@
+# Adding a New Exchange to Aggr
+
+This guide provides a comprehensive step-by-step process for adding a new cryptocurrency exchange to the Aggr trading aggregator, based on the WhiteBIT integration (PR #429).
+
+## Prerequisites
+
+Before starting, ensure the exchange meets these requirements:
+
+1. **Public WebSocket API** - The exchange must provide a public WebSocket API for real-time trade data
+2. **Public REST API** - Must have a markets/products endpoint to fetch available trading pairs
+3. **No Authentication Required** - The APIs should be accessible without API keys for public market data
+4. **Proper Documentation** - Clear API documentation for both REST and WebSocket endpoints
+
+## Step-by-Step Process
+
+### 1. Research and Validation
+
+1. **Study the Exchange API Documentation**
+   - Identify the WebSocket endpoint URL
+   - Find the REST API endpoint for markets/products
+   - Understand the message format for trade subscriptions
+   - Check if the API requires any special headers or parameters
+
+2. **Test the APIs**
+   - Verify the REST endpoint returns market data
+   - Test WebSocket connection and trade subscription
+   - Ensure the APIs are publicly accessible (no CORS issues)
+
+### 2. Asset Preparation
+
+1. **Download Exchange Assets**
+   - Get the exchange logo/icon in high quality
+   - Preferred format: SVG or high-resolution PNG
+
+2. **Convert to Icon Font Format**
+   - Use [IcoMoon.io](https://icomoon.io/) to convert the logo to SVG format
+   - Ensure the SVG is optimized and clean
+   - The icon should be recognizable at small sizes
+
+### 3. Code Implementation
+
+#### 3.1 Create the Exchange Class
+
+Create a new file: `src/worker/exchanges/{exchangename}.ts`
+
+```typescript
+import Exchange from '../exchange'
+
+export default class EXCHANGENAME extends Exchange {
+  id = 'EXCHANGENAME'
+
+  protected endpoints = {
+    PRODUCTS: 'https://api.exchange.com/v1/markets'
```

Comment:
Some exchanges provide separate endpoints for SPOT and PERPS.
Both endpoints are called simultaneously, and their responses are passed as arguments to the NewExchange class method formatProducts.